components/brave_ads/core/internal/history/ad_history_manager.cc

```diff
@@ -195,11 +207,12 @@ bool AdHistoryManager::ToggleSaveAd(
     NotifyDidUnsaveAd(mutable_ad_history_item);
   }
 
-  return mutable_ad_history_item.is_saved;
+  std::move(callback).Run(/*success=*/true);
```

Comment:
This is a prerequisite for future work to transition from JSON to a database. That work will then return either true or false, instead of always true.